### PR TITLE
Add Top Sellers carousel

### LINKED ISSUE #3: Add "top sellers" section  📝🎨

### PR DESCRIPTION
**Pull request by BeJuanitox**
## Before
https://app.claap.io/developers-inc/c-CU6RMp4tEE-6fpDu0rJQGAO

## After
image

Closes #3

### CODE REVIEW

File: src/app/components/top-sellers/top-sellers.component.html [hunk 1]
```diff
@@ -0,0 +1,22 @@
+<h1 class="titleTopSellers">Más vendidos</h1>
```
Comment (Andres2D) [inline src/app/components/top-sellers/top-sellers.component.html]:
```suggestion
<h1 class="title">Más vendidos</h1>
```

File: src/app/components/top-sellers/top-sellers.component.html [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+<div *ngIf="productSellers && productSellers.length > 0" class="carousel-container">
+  <button *ngIf="controls" class="btn-carousel" (click)="onPrevClick()">
```
Comment (Andres2D) [inline src/app/components/top-sellers/top-sellers.component.html]: Don't use contractions.
```suggestion
  <button *ngIf="controls" class="btn-carousel" (click)="onPreviousClick()">
```

File: src/app/components/top-sellers/top-sellers.component.html [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    <i class="fa-solid fa-chevron-left"></i>
+  </button>
+  <div class="globalContainer">
+    <div *ngFor="let prod of productSellers; let i=index" [ngClass]="{'image-active': selectedIndex === i}"
```
Comment (Andres2D) [inline src/app/components/top-sellers/top-sellers.component.html]: Same here.
```suggestion
    <div *ngFor="let product of productSellers; let i=index" [ngClass]="{'image-active': selectedIndex === i}"
```

File: src/app/components/top-sellers/top-sellers.component.html [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+      class="productTopSellers">
+      <h1> {{prod.name | titlecase}} </h1>
+      <img [src]="prod.image" alt="ProductName" class="productImage">
```
Comment (Andres2D) [inline src/app/components/top-sellers/top-sellers.component.html]: The alt image shouldn't be repeated.
```suggestion
      <img [src]="prod.image" [alt]="prod.image" class="productImage">
```

Comment (Andres2D) [inline src/app/components/top-sellers/top-sellers.component.html]: Sometime you'r using `cammelCase` and other times you're using` kebab-case` for the classes names.
Please keep with a single standar.
```suggestion
      <img [src]="prod.image" alt="ProductName" class="product-image">
```

File: src/app/components/top-sellers/top-sellers.component.html [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+      <h2> {{prod.price}} MXN</h2>
```
Comment (Andres2D) [inline src/app/components/top-sellers/top-sellers.component.html]: Use the `currency` pipe: https://angular.io/api/common/CurrencyPipe#usage-notes

File: src/app/components/top-sellers/top-sellers.component.scss [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+  opacity: 0.9;
+}
+
+.btn-carousel:active {
+  opacity: 0.5;
+}
+
+.fa-solid {
+  font-size: 50px;
+}
```
Comment (Andres2D) [inline src/app/components/top-sellers/top-sellers.component.scss:107]: Use the nesting feature of sass on this file.

File: src/app/components/top-sellers/top-sellers.component.ts [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+  templateUrl: './top-sellers.component.html',
+  styleUrls: ['./top-sellers.component.scss']
+})
+export class TopSellersComponent implements OnInit {
+
+  @Input() productSellers: TopSellers[] = mockTopSellers;
+  @Input() indicators = true;
+  @Input() controls = true;
+  @Input() autoSlide = false;
+  @Input() slideInterval = 3000;
```
Comment (Andres2D) [inline src/app/components/top-sellers/top-sellers.component.ts]:
```suggestion
  @Input() indicators: boolean = true;
  @Input() controls: boolean = true;
  @Input() autoSlide: boolean = false;
  @Input() slideInterval: number = 3000;
```

File: src/app/components/top-sellers/top-sellers.component.ts [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+
+  selectedIndex = 0;
```
Comment (Andres2D) [inline src/app/components/top-sellers/top-sellers.component.ts]:
```suggestion
  selectedIndex: number = 0;
```

File: src/app/components/top-sellers/top-sellers.component.ts [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+
+  selectedProducts(index: number): void {
+    this.selectedIndex = index; 
+  }
+
+  onPrevClick(): void {
+    if (this.selectedIndex === 0) {
+      this.selectedIndex = this.productSellers.length - 1;
+    } else {
+      this.selectedIndex--;
```
Comment (Andres2D) [inline src/app/components/top-sellers/top-sellers.component.ts:40]: Is this is actually working?

File: src/app/mock/sellers.ts [hunk 10]
```diff
@@ -0,0 +1,32 @@
+export const mockTopSellers = [
```
Comment (Andres2D) [inline src/app/mock/sellers.ts]:
```suggestion
export const mockTopSellers: TopSellers[] = [
```

**Review by Andres2D — CHANGES_REQUESTED** [review]
In overall it was a great implementation, you're using a lot of key angular features.

File: src/app/components/top-sellers/top-sellers.component.html (hunk 4 above)
Comment (BeJuanitox) [reply src/app/components/top-sellers/top-sellers.component.html]: prod.name

File: src/app/components/top-sellers/top-sellers.component.html [hunk 11]
```diff
@@ -0,0 +1,24 @@
+<div class="title-container">
+  <h1 class="title">Más vendidos</h1>
+</div>
```
Comment (Andres2D) [inline src/app/components/top-sellers/top-sellers.component.html:3]: Why do you need a container for a header?

**Review by Andres2D — APPROVED** [review]

File: src/app/components/top-sellers/top-sellers.component.html (hunk 11 above)
Comment (BeJuanitox) [reply src/app/components/top-sellers/top-sellers.component.html:3]: To easily center text

**Pull request merged**